### ISSUE TITLE
comparison of Danger::Violation with Danger::Violation failed (ArgumentError)

### ISSUE THREAD

**Issue by Nef10** [danger/danger#924]
# Report

## What did you do?

Ran `danger(fail_on_errors: true)` in fastlane on Travis

## What did you expect to happen?

Running my Dangerfile without crashing.

## What happened instead?

```
[23:05:04]: ▸ Coverage report successfully created at path: /Users/travis/build/Nef10/MyProjectName/fastlane/xcov_report/tmp/report.json20171118-3166-9l7ts0
/Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/request_sources/github/github.rb:503:in `sort': comparison of Danger::Violation with Danger::Violation failed (ArgumentError)
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/request_sources/github/github.rb:503:in `inline_violations_group'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/request_sources/github/github.rb:163:in `update_pull_request!'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/danger_core/dangerfile.rb:248:in `post_results'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/danger_core/dangerfile.rb:278:in `run'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/danger_core/executor.rb:27:in `run'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/commands/runner.rb:66:in `run'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/bin/danger:5:in `<top (required)>'
	from /Users/travis/.rvm/gems/ruby-2.4.2/bin/danger:23:in `load'
	from /Users/travis/.rvm/gems/ruby-2.4.2/bin/danger:23:in `<main>'
	from /Users/travis/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `eval'
	from /Users/travis/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `<main>'
```

<details><summary>Full log</summary><p>

```
[23:04:59]: --------------------
[23:04:59]: --- Step: danger ---
[23:04:59]: --------------------
[23:04:59]: $ bundle exec danger --fail-on-errors=true
[23:05:01]: ▸ [23:05:01]: Skipping file blacklisting as no ignore file was found at path ./fastlane/.xcovignore
[23:05:01]: ▸ [23:05:01]: $ xcodebuild -list -project ./MyProjectName.xcodeproj
[23:05:02]: ▸ [23:05:02]: Skipping file blacklisting as no ignore file was found at path ./fastlane/.xcovignore
[23:05:02]: ▸ +-----------------------------+---------------------------------+
[23:05:02]: ▸ |                    Summary for xcov 1.3.0                     |
[23:05:02]: ▸ +-----------------------------+---------------------------------+
[23:05:02]: ▸ | minimum_coverage_percentage | 100.0                           |
[23:05:02]: ▸ | project                     | ./MyProjectName.xcodeproj |
[23:05:02]: ▸ | scheme                      | MyProjectName             |
[23:05:02]: ▸ | output_directory            | ./fastlane/xcov_report          |
[23:05:02]: ▸ | html_report                 | true                            |
[23:05:02]: ▸ | markdown_report             | false                           |
[23:05:02]: ▸ | json_report                 | false                           |
[23:05:02]: ▸ | skip_slack                  | false                           |
[23:05:02]: ▸ | slack_username              | xcov                            |
[23:05:02]: ▸ | slack_message               | Your *xcov* coverage report     |
[23:05:02]: ▸ | ignore_file_path            | ./fastlane/.xcovignore          |
[23:05:02]: ▸ | include_test_targets        | false                           |
[23:05:02]: ▸ | only_project_targets        | false                           |
[23:05:02]: ▸ +-----------------------------+---------------------------------+
[23:05:03]: ▸ [23:05:03]: $ xcodebuild -showBuildSettings -scheme MyProjectName -project ./MyProjectName.xcodeproj
[23:05:04]: ▸ [23:05:04]: $ /Users/travis/.rvm/gems/ruby-2.4.2/gems/xcov-1.3.0/lib/xcov-core/bin/xcov-core -s /Users/travis/Library/Developer/Xcode/DerivedData/MyProjectName-eytejglpoczrzqgcjkiqsyhwtlgs/Logs/Test/9BE8513A-F2D3-4130-8427-2A8C96B0E8C3.xccoverage -o ./fastlane/xcov_report/tmp/report.json20171118-3166-9l7ts0 --include-lines-info
[23:05:04]: [23:05:04]: ▸ Loading...
[23:05:04]: [23:05:04]: ▸ ------ xcov-core ------
[23:05:04]: [23:05:04]: ▸ Opening .xccoverage file at path: /Users/travis/Library/Developer/Xcode/DerivedData/MyProjectName-eytejglpoczrzqgcjkiqsyhwtlgs/Logs/Test/9BE8513A-F2D3-4130-8427-2A8C96B0E8C3.xccoverage
[23:05:04]: [23:05:04]: ▸ Parsing .xccoverage file...
[23:05:04]: [23:05:04]: ▸ File successfully parsed
[23:05:04]: [23:05:04]: ▸ Serializing coverage report...
[23:05:04]: [23:05:04]: ▸ Report successfully serialized
[23:05:04]: [23:05:04]: ▸ Writing report on disk...
[23:05:04]: [23:05:04]: ▸ Coverage report successfully created at path: /Users/travis/build/Nef10/MyProjectName/fastlane/xcov_report/tmp/report.json20171118-3166-9l7ts0
[23:05:04]: ▸ /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/request_sources/github/github.rb:503:in `sort': comparison of Danger::Violation with Danger::Violation failed (ArgumentError)
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/request_sources/github/github.rb:503:in `inline_violations_group'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/request_sources/github/github.rb:163:in `update_pull_request!'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/danger_core/dangerfile.rb:248:in `post_results'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/danger_core/dangerfile.rb:278:in `run'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/danger_core/executor.rb:27:in `run'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/commands/runner.rb:66:in `run'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/bin/danger:5:in `<top (required)>'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/bin/danger:23:in `load'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/bin/danger:23:in `<main>'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `eval'
[23:05:04]: ▸ from /Users/travis/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `<main>'
+----------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                              Lane Context                                                                                              |
+----------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM           | mac                                                                                                                                                                       |
| PLATFORM_NAME              | mac                                                                                                                                                                       |
| LANE_NAME                  | mac ci                                                                                                                                                                    |
| SCAN_DERIVED_DATA_PATH     | /Users/travis/Library/Developer/Xcode/DerivedData/MyProjectName-eytejglpoczrzqgcjkiqsyhwtlgs                                                                        |
| SCAN_GENERATED_PLIST_FILES | ["/Users/travis/Library/Developer/Xcode/DerivedData/MyProjectName-eytejglpoczrzqgcjkiqsyhwtlgs/Logs/Test/9BE8513A-F2D3-4130-8427-2A8C96B0E8C3_TestSummaries.plist"] |
| SCAN_GENERATED_PLIST_FILE  | /Users/travis/Library/Developer/Xcode/DerivedData/MyProjectName-eytejglpoczrzqgcjkiqsyhwtlgs/Logs/Test/9BE8513A-F2D3-4130-8427-2A8C96B0E8C3_TestSummaries.plist     |
+----------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
[23:05:04]: Exit status of command 'bundle exec danger --fail-on-errors=true' was 1 instead of 0.
[23:05:01]: Skipping file blacklisting as no ignore file was found at path ./fastlane/.xcovignore
[23:05:01]: $ xcodebuild -list -project ./MyProjectName.xcodeproj
[23:05:02]: Skipping file blacklisting as no ignore file was found at path ./fastlane/.xcovignore
+-----------------------------+---------------------------------+
|                    Summary for xcov 1.3.0                     |
+-----------------------------+---------------------------------+
| minimum_coverage_percentage | 100.0                           |
| project                     | ./MyProjectName.xcodeproj |
| scheme                      | MyProjectName             |
| output_directory            | ./fastlane/xcov_report          |
| html_report                 | true                            |
| markdown_report             | false                           |
| json_report                 | false                           |
| skip_slack                  | false                           |
| slack_username              | xcov                            |
| slack_message               | Your *xcov* coverage report     |
| ignore_file_path            | ./fastlane/.xcovignore          |
| include_test_targets        | false                           |
| only_project_targets        | false                           |
+-----------------------------+---------------------------------+
[23:05:03]: $ xcodebuild -showBuildSettings -scheme MyProjectName -project ./MyProjectName.xcodeproj
[23:05:04]: $ /Users/travis/.rvm/gems/ruby-2.4.2/gems/xcov-1.3.0/lib/xcov-core/bin/xcov-core -s /Users/travis/Library/Developer/Xcode/DerivedData/MyProjectName-eytejglpoczrzqgcjkiqsyhwtlgs/Logs/Test/9BE8513A-F2D3-4130-8427-2A8C96B0E8C3.xccoverage -o ./fastlane/xcov_report/tmp/report.json20171118-3166-9l7ts0 --include-lines-info
[23:05:04]: ▸ Loading...
[23:05:04]: ▸ ------ xcov-core ------
[23:05:04]: ▸ Opening .xccoverage file at path: /Users/travis/Library/Developer/Xcode/DerivedData/MyProjectName-eytejglpoczrzqgcjkiqsyhwtlgs/Logs/Test/9BE8513A-F2D3-4130-8427-2A8C96B0E8C3.xccoverage
[23:05:04]: ▸ Parsing .xccoverage file...
[23:05:04]: ▸ File successfully parsed
[23:05:04]: ▸ Serializing coverage report...
[23:05:04]: ▸ Report successfully serialized
[23:05:04]: ▸ Writing report on disk...
[23:05:04]: ▸ Coverage report successfully created at path: /Users/travis/build/Nef10/MyProjectName/fastlane/xcov_report/tmp/report.json20171118-3166-9l7ts0
/Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/request_sources/github/github.rb:503:in `sort': comparison of Danger::Violation with Danger::Violation failed (ArgumentError)
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/request_sources/github/github.rb:503:in `inline_violations_group'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/request_sources/github/github.rb:163:in `update_pull_request!'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/danger_core/dangerfile.rb:248:in `post_results'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/danger_core/dangerfile.rb:278:in `run'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/danger_core/executor.rb:27:in `run'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/lib/danger/commands/runner.rb:66:in `run'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
	from /Users/travis/.rvm/gems/ruby-2.4.2/gems/danger-5.5.5/bin/danger:5:in `<top (required)>'
	from /Users/travis/.rvm/gems/ruby-2.4.2/bin/danger:23:in `load'
	from /Users/travis/.rvm/gems/ruby-2.4.2/bin/danger:23:in `<main>'
	from /Users/travis/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `eval'
	from /Users/travis/.rvm/gems/ruby-2.4.2/bin/ruby_executable_hooks:15:in `<main>'
```

</p></details>

## Your Environment

* Which CI are you running on? Travis
* Are you running the latest version of Danger? Yes, `5.5.5`
* What is your Dangerfile?

  ```ruby
  xcov.report(minimum_coverage_percentage: 100.00)
  xcode_summary.report 'build/reports/errors.json'
  jazzy.check fail: :all
  ```

**Comment by Nef10** [comment]
I could track it down to [danger-jazzy](https://github.com/fwal/danger-jazzy) when it has warnings or errors to show

**Comment by misterwell** [comment]
Any ideas on how to resolve this? Is it an issue within danger-jazzy or danger itself?

**Comment by orta** [comment]
Tough to say TBH - Danger isn't doing [anything fancy here](https://github.com/danger/danger/blob/master/lib/danger/request_sources/github/github.rb#L500-L507) ( it's crashing in ordering ) 

Without an easy repo case so that someone can pull out what warnings, errors, messages and markdowns look like it's not obvious where the error lays

**Comment by Nef10** [comment]
I tried to create a simple project to reproduce but there everything works: https://github.com/Nef10/Danger-924/pull/2

**Comment by Nef10** [comment]
It seems to be related to either some violations having line as null or that some violations are duplicated. I fixed all these violations and it is no longer crashing.

See realm/jazzy#936

**Comment by orta** [comment]
Very cool, Danger also got stronger equality checking in the last version update that might help against some of this

**Comment by GRiMe2D** [comment]
I'm currently having same crash.

Output of `bundle exec danger pr <pr_url>`
```
Warnings:
- [ ] app/src/main/java/PreferencesHelper.java#L8: 
Avoid unused imports such as 'com.google.gson.Gson'
- [ ] app/src/main/java/PreferencesHelper.java#L9: 
Avoid unused imports such as 'com.google.gson.reflect.TypeToken'
- [ ] app/src/main/java/PreferencesHelper.java#L15: 
Avoid unused imports such as 'java.util.List'
- [ ] app/src/main/java/PreferencesHelper.java#L25: 
Avoid unused imports such as 'Place'
- [ ] app/src/main/java/PreferencesHelper.java#L136: 
These nested if statements could be combined
- [ ] app/src/main/java/PreferencesHelper.java#L5: 
Avoid unused imports such as 'android.util.Log'
- [ ] app/src/main/java/PlaceAddMapActivity.java#L8: 
Avoid unused imports such as 'com.google.gson.Gson'
- [ ] app/src/main/java/PlaceAddMapActivity.java#L9: 
Avoid unused imports such as 'com.google.gson.reflect.TypeToken'
- [ ] app/src/main/java/PlaceAddMapActivity.java#L15: 
Avoid unused imports such as 'java.util.List'
- [ ] app/src/main/java/PlaceAddMapActivity.java#L25: 
Avoid unused imports such as 'Place'
- [ ] app/src/main/java/PlaceAddMapActivity.java#L136: 
These nested if statements could be combined
- [ ] app/src/main/java/PlaceAddMapActivity.java#L5: 
Avoid unused imports such as 'android.util.Log'
- [ ] app/src/main/java/PlacesAdapter.java#L8: 
Avoid unused imports such as 'com.google.gson.Gson'
- [ ] app/src/main/java/PlacesAdapter.java#L9: 
Avoid unused imports such as 'com.google.gson.reflect.TypeToken'
- [ ] app/src/main/java/PlacesAdapter.java#L15: 
Avoid unused imports such as 'java.util.List'
- [ ] app/src/main/java/PlacesAdapter.java#L25: 
Avoid unused imports such as 'Place'
- [ ] app/src/main/java/PlacesAdapter.java#L136: 
These nested if statements could be combined
- [ ] app/src/main/java/PlacesAdapter.java#L5: 
Avoid unused imports such as 'android.util.Log'
- [ ] app/src/main/java/DataManager.java#L829: Line is longer than 130 characters (found 144).
- [ ] app/src/main/java/MapActivity.java#L353: WhitespaceAround: '{' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
- [ ] app/src/main/java/MapActivity.java#L353: WhitespaceAround: '}' is not preceded with whitespace.
- [ ] app/src/main/java/PlaceAddMapActivity.java#L135: WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
- [ ] app/src/main/java/PlaceAddMapActivity.java#L136: WhitespaceAround: 'if' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
- [ ] app/src/main/java/PlaceAddMapActivity.java#L290: WhitespaceAround: '{' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
- [ ] app/src/main/java/PlaceAddMapActivity.java#L290: WhitespaceAround: '}' is not preceded with whitespace.
- [ ] app/src/main/java/PlaceAddMapActivity.java#L293: WhitespaceAround: '{' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
- [ ] app/src/main/java/PlaceAddMapActivity.java#L293: WhitespaceAround: '}' is not preceded with whitespace.
- [ ] app/src/main/java/PlaceAddMapPresenter.java#L51: WhitespaceAround: 'for' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)
- [ ] app/src/main/java/PlacesAdapter.java#L65: Line is longer than 130 characters (found 131).
- [ ] app/src/main/java/MapUtils.java#L201: WhitespaceAround: '{' is not preceded with whitespace.
```

Output when in CI
```
bundler: failed to load command: danger (Vendor/Bundle/ruby/2.3.0/bin/danger)
ArgumentError: comparison of Danger::Violation with Danger::Violation failed
  Vendor/Bundle/ruby/2.3.0/gems/danger-5.5.10/lib/danger/request_sources/github/github.rb:502:in `sort'
  Vendor/Bundle/ruby/2.3.0/gems/danger-5.5.10/lib/danger/request_sources/github/github.rb:502:in `inline_violations_group'
  Vendor/Bundle/ruby/2.3.0/gems/danger-5.5.10/lib/danger/request_sources/github/github.rb:163:in `update_pull_request!'
  Vendor/Bundle/ruby/2.3.0/gems/danger-5.5.10/lib/danger/danger_core/dangerfile.rb:248:in `post_results'
  Vendor/Bundle/ruby/2.3.0/gems/danger-5.5.10/lib/danger/danger_core/dangerfile.rb:279:in `run'
  Vendor/Bundle/ruby/2.3.0/gems/danger-5.5.10/lib/danger/danger_core/executor.rb:28:in `run'
  Vendor/Bundle/ruby/2.3.0/gems/danger-5.5.10/lib/danger/commands/runner.rb:68:in `run'
  Vendor/Bundle/ruby/2.3.0/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
  Vendor/Bundle/ruby/2.3.0/gems/danger-5.5.10/bin/danger:5:in `<top (required)>'
  Vendor/Bundle/ruby/2.3.0/bin/danger:22:in `load'
  Vendor/Bundle/ruby/2.3.0/bin/danger:22:in `<top (required)>'
```

**Comment by fwal** [comment]
Addressing the original bug in #978.

Released version [1.1.0](https://github.com/fwal/danger-jazzy/releases/tag/v1.1.0) of danger-jazzy that works around this issue.

**Comment by Nef10** [comment]
@fwal thanks!
